dali/util/odirect_file.cc

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2017-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <errno.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <cstdio>
+#include <cstring>
+#include <memory>
+#include <string>
+
+#include "dali/core/util.h"
+#include "dali/util/odirect_file.h"
+#include "dali/core/error_handling.h"
+
+namespace dali {
+
+static constexpr size_t kODirectAlignment = 4096;
+static constexpr size_t kODirectChunkSize = 2 << 20;  // 2M
+
+static size_t GetODirectEnvVals(const char *env_name, size_t min_value, size_t default_value,
+                                size_t alignment = 0) {
```

Comment:
```suggestion
                                size_t alignment = 1) {
```
Wouldn't it work better? Every number is a multiple of 1...